### PR TITLE
Bump up API version to v1beta3

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: pkg/apis/direct.csi.min.io/v1beta3/types.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	Status DirectCSIDriveStatus `json:"status,omitempty"`
+}
+
+type DirectCSIDriveSpec struct {
+	// +optional
+	RequestedFormat *RequestedFormat `json:"requestedFormat,omitempty"`
+	// required
+	DirectCSIOwned bool `json:"directCSIOwned"`
+	// +optional
+	DriveTaint map[string]string `json:"driveTaint,omitempty"`
```
Comment (Praveenrajmani) [inline pkg/apis/direct.csi.min.io/v1beta3/types.go:52]: Can we remove this field as we don't use it? We can also do this in a separate PR

cc @wlan0

**Comment by Praveenrajmani** [comment]
As discussed, we need to add another field for udev FSUUID as it is different with what we read from superblock.

File: pkg/utils/drive.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		if strings.HasPrefix(mountPoint, "/var/lib/direct-csi/") {
+			return true
+		}
+	}
+	return false
+}
+
+func NewDirectCSIDriveStatus(device *sys.Device, nodeID string, topology map[string]string) directcsi.DirectCSIDriveStatus {
+	driveStatus := directcsi.DriveStatusAvailable
+	if device.ReadOnly || device.Partitioned || device.SwapOn || device.Master != "" || !isDirectCSIMount(device.MountPoints) {
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply pkg/utils/drive.go:50]: Devices those mounted beyond `direct-csi` scope, what value we get to know them as `Available` and later showing error message `No such file or directory`?

I would say this check is a fix to problem currently we show `No such file or directory` with `Available` state which doesn't give any value to users.

File: pkg/apis/direct.csi.min.io/v1beta2/types.go [hunk 3]
```diff
@@ -31,7 +31,6 @@ const (
 // +genclient
 // +genclient:nonNamespaced
 // +kubebuilder:resource:scope=Cluster
-// +kubebuilder:storageversion
```
Comment (wlan0) [inline pkg/apis/direct.csi.min.io/v1beta2/types.go:34]: @Praveenrajmani can you verify this

**Review by wlan0 — DISMISSED** [review]

File: pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+	out.SerialNumber = in.SerialNumber
+	out.TotalCapacity = in.TotalCapacity
+	out.PhysicalBlockSize = in.PhysicalBlockSize
+	out.LogicalBlockSize = in.LogicalBlockSize
+	out.Topology = *(*map[string]string)(unsafe.Pointer(&in.Topology))
+	out.AccessTier = v1beta2.AccessTier(in.AccessTier)
+	// INFO: in.FilesystemUUID opted out of conversion generation
+	// INFO: in.PartitionUUID opted out of conversion generation
+	// INFO: in.MajorNumber opted out of conversion generation
+	// INFO: in.MinorNumber opted out of conversion generation
```
Comment (Praveenrajmani) [inline pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go:242]: Looks like these fields are opted out for conversion, We need to remove the `+k8s:conversion-gen=false` comment on these fields in `pkg/apis/direct.csi.min.io/v1beta3/types.go`

File: pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+	out.SerialNumber = in.SerialNumber
+	out.TotalCapacity = in.TotalCapacity
+	out.PhysicalBlockSize = in.PhysicalBlockSize
+	out.LogicalBlockSize = in.LogicalBlockSize
+	out.Topology = *(*map[string]string)(unsafe.Pointer(&in.Topology))
+	out.AccessTier = AccessTier(in.AccessTier)
+	// INFO: in.FilesystemUUID opted out of conversion generation
+	// INFO: in.PartitionUUID opted out of conversion generation
+	// INFO: in.MajorNumber opted out of conversion generation
+	// INFO: in.MinorNumber opted out of conversion generation
```
Comment (Praveenrajmani) [inline pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go:287]: Looks like these fields are opted out for conversion, We need to remove the `+k8s:conversion-gen=false` comment on these fields in `pkg/apis/direct.csi.min.io/v1beta2/types.go`

**Review by Praveenrajmani — COMMENTED** [review]
Some minor fixes required in conversion generated functions. Few fields were opted out for conversion.

Also, the `--input-dirs` for `conversion-gen` should point to the latest pkg added.

File: pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go (hunk 5 above)
Comment (balamurugana) [reply pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go]: Done

File: pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go (hunk 4 above)
Comment (balamurugana) [reply pkg/apis/direct.csi.min.io/v1beta3/conversion_generated.go]: Done

**Review by Praveenrajmani — APPROVED** [review]
